### PR TITLE
Add weights_only=True in torch.load

### PR DESCRIPTION
**Pull request by terry-for-github**
FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.

### CODE REVIEW

**Comment by terry-for-github** [comment]
I didnt check which case should not use weights_only=True. I just add it in all of the torch.load. Hope it will help.

**Review by tjruwase — APPROVED** [review]

**Comment by tjruwase** [comment]
> I didnt check which case should not use weights_only=True. I just add it in all of the torch.load. Hope it will help.

@terry-for-github, thanks for making DeepSpeed more secure.

File: deepspeed/checkpoint/deepspeed_checkpoint.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def show_transformer_file_map(self):
         self._dump_mapping(self.transformer_file_map, 'rank_to_transformer_files')
 
     def _build_global_state(self):
-        sd = torch.load(self.mp_rank_files[0], map_location=torch.device('cpu'))
```
Comment (loadams) [inline deepspeed/checkpoint/deepspeed_checkpoint.py:119]: Hi @terry-for-github - could you run the pre-commit formatter, that should fix the formatting errors on these files.

Comment (terry-for-github) [reply deepspeed/checkpoint/deepspeed_checkpoint.py:119]: ok, I'll try that.

**Comment by terry-for-github** [comment]
Is this OK? I haven't use pre-commit tools brefore. I do it by myself lol.

**Comment by loadams** [comment]
> Is this OK? I haven't use pre-commit tools brefore. I do it by myself lol.

Yes @terry-for-github thanks, the formatting checks appear to be passing, though other checks are not now.

**Comment by terry-for-github** [comment]
I found that not all of the `torch.load` are compatible with `weights_only=True`. Some of them should be set to `weights_only=False` explicitly instead.

**Comment by loadams** [comment]
> I found that not all of the `torch.load` are compatible with `weights_only=True`. Some of them should be set to `weights_only=False` explicitly instead.

Agreed, we've fixed most of the unrelated CI issues, would you be interested in continuing this PR?

**Comment by loadams** [comment]
> > I found that not all of the `torch.load` are compatible with `weights_only=True`. Some of them should be set to `weights_only=False` explicitly instead.
> 
> Agreed, we've fixed most of the unrelated CI issues, would you be interested in continuing this PR?

Hi @terry-for-github - no worries if you aren't, but I'll make a PR as well to fix these warnings.

**Comment by loadams** [comment]
Closing in favor of #6751

**Pull request closed without merging**